compiler/compile/VirtualGuard.cpp

```diff
@@ -134,22 +134,35 @@ TR_VirtualGuard::createRubyInlineGuard
 (TR_VirtualGuardKind kind, TR::Compilation * comp, int16_t calleeIndex,
  TR::Node* callNode, TR::TreeTop * destination, TR_OpaqueClassBlock *thisClass)
 {
+#ifdef RUBY_PROJECT_SPECIFIC
   //Ensure that ILGen hasn't changed the children's layout we expect.
-  TR_ASSERT(  ((callNode->getNumChildren() == 3) &&
+  TR_ASSERT(  ((callNode->getNumChildren() == 4) &&
        callNode->getSecondChild() &&
        callNode->getSecondChild()->getOpCodeValue() == TR::aconst), "Unexpected children in hierarchy of vm_send_without_block when creating ruby inline guard.");
 
-  TR::Node* receiver = callNode->getThirdChild();
-  TR::Node* callInfo = callNode->getSecondChild();
+  TR::Node* receiver  = callNode->getChild(3);
+  TR::Node* callCache = callNode->getThirdChild();
+  CALL_CACHE cc = reinterpret_cast<CALL_CACHE>(callCache->getAddress());
+
 
   //Call the ruby inline guard routine.
   TR::SymbolReferenceTable *symRefTab = comp->getSymRefTab();
   TR::SymbolReference *inline_guard_helperSymRef = symRefTab->findOrCreateRuntimeHelper(RubyHelper_vm_send_woblock_inlineable_guard, true, true, false);
 
-  TR::Node* isInlineable = TR::Node::create(TR::lcall, 2);
+  TR::Node* isInlineable = TR::Node::create(TR::lcall, 3);
+
+  // Nail down what the serial numbers are at time of inlining -- these will 
+  // match the inlined body serial numbers. 
   isInlineable->setSymbolReference(inline_guard_helperSymRef);
-  isInlineable->setAndIncChild(0, TR::Node::aconst(callInfo, (uintptrj_t)callInfo->getAddress()));
-  isInlineable->setAndIncChild(1, receiver);
+
+  //TODO: These trees ought to be using i|l consts... depending on the platform size, 
```

Comment:
The general policy with OMR has been to not introduce TODO's into the code.  If you want to track outstanding work, please create an issue so it doesn't get forgotten.